mysql-test/suite/rocksdb/t/collation.test

```diff
@@ -166,16 +96,76 @@ CREATE TABLE c (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rock
 DROP TABLE a, b;
 
 # test invalid regex (trailing escape)
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
 SET GLOBAL rocksdb_strict_collation_exceptions="abc\\";
---exec grep "Invalid pattern" $MYSQLTEST_VARDIR/log/mysqld.1.err | cut -d] -f2
+--let SEARCH_PATTERN=Invalid pattern in strict_collation_exceptions: abc\\\\
+--source include/search_pattern_in_file.inc
+
 --error ER_UNKNOWN_ERROR
 CREATE TABLE abc (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+--error ER_UNKNOWN_ERROR
+CREATE TABLE ABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
 SET GLOBAL rocksdb_strict_collation_exceptions="abc";
 CREATE TABLE abc (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+DROP TABLE abc;
+--error ER_UNKNOWN_ERROR
+CREATE TABLE ABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
+SET GLOBAL rocksdb_strict_collation_exceptions="ABC";
+--error ER_UNKNOWN_ERROR
+CREATE TABLE abc (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+CREATE TABLE ABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+DROP TABLE ABC;
+
+--error ER_UNKNOWN_ERROR
+CREATE TABLE bcd (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
+SET GLOBAL rocksdb_strict_collation_exceptions="^ABC";
 --error ER_UNKNOWN_ERROR
 CREATE TABLE abcd (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+CREATE TABLE ABCD (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+DROP TABLE ABCD;
+
+--error ER_UNKNOWN_ERROR
+CREATE TABLE ZABC (id INT PRIMARY KEY, value varchar(50), index(value)) engine=rocksdb charset utf8;
+
+
+# test with server comparing in lower case
+--let $restart_parameters="restart: --lower_case_table_names=1"
```

Comment:
Again though, I bent a little to cover some of this, but I still think this is not appropriate to place the testing of what is basically a new sysvar type within a test for a this specific option. That implies then that we need to replicate all the test work we do here for every variable that uses this functionality. I am not at all opposed to testing but it must be done in the correct way and in the correct place otherwise we end up with more test fragmentation. By your suggestion, I now need a collation, collation_lctn_0, collation_lctn_1, and ideally a collation_lctn_2 test, all of which exist solely to test regex and char case which has nothing to do with the functionality of the strict collation check. We will also now need the same for the existing rocksdb read_free_rpl_tables since it uses the same facility (and eventually to be moved up into SQL layer as discussed in MYR-6), and we will need to continue the redundancy to every and all options that use this Regex facility. I do not believe that is at all reasonable. That being said, I can not offer an alternative as there is no other mtr accessible UI to test this specific regex functionality.